app/main.py

```diff
@@ -1,34 +1,210 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self, start: tuple,
+                 end: tuple, is_drowned: bool = False) -> None:
+        self.is_drowned = is_drowned
+        self.decks = []
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        if start[0] == start[1] and end[0] == end[1]:
+            self.start_column, self.end_column = start[1], end[1]
+            self.end_row, self.start_row = end[0], start[0]
+            self.decks.append(Deck(self.start_row, self.end_column))
+            self.position = "single"
+        elif start[0] == end[0]:
+            self.position = "horizontal"
+            row = start[0]
+            for column in range(start[1], end[1] + 1):
+                self.decks.append(Deck(row, column))
+            self.start_column, self.end_column = start[1], end[1]
+        elif start[1] == end[1]:
+            self.position = "vertical"
+            column = start[1]
+            for row in range(start[0], end[0] + 1):
+                self.decks.append(Deck(row, column))
+            self.start_row, self.end_row = start[0], end[0]
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def __str__(self) -> str:
+        ship = ""
+        if self.position == "single":
+            ship += "□"
+        if self.position == "horizontal":
+            for _ in range(len(self.decks)):
+                ship += "□  "
+
+        if self.position == "vertical":
+            for _ in range(len(self.decks)):
+                ship += "□\n"
+
+        return f"\n{ship}\n"
+
+    def get_deck(self, row: int, column: int) -> Deck:
+        for deck in self.decks:
+            if (row, column) == (deck.row, deck.column):
+                return deck
+
+    def fire(self, row: int, column: int) -> str:
+        for deck in self.decks:
+            if (row, column) == (deck.row, deck.column):
+                deck.is_alive = False
+        if True not in [deck.is_alive for deck in self.decks]:
```

Comment:
The condition `if True not in [deck.is_alive for deck in self.decks]:` is correct but could be simplified to `if all(not deck.is_alive for deck in self.decks):` for better readability.